src/main/java/dataloader/JsonDataLoader.java

```diff
@@ -0,0 +1,37 @@
+package dataloader;
+
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Simple Class that parses data set from a json file
+ * without knowing the file's format
+ * & prints out the key value pairs.
+ */
+
+
+public class JsonDataLoader {
+
+    public void parseFile(String jsonFile) throws IOException {
+
+        String location = "/path/to/file.json";
```

Comment:
Similarly here, we don't need the ```location``` variable. This is specified in the ```jsonFile``` parameter

Reply:
provide an example here, please.

Reply:
well the input ```jsonFile``` may also contain the path: 
```
String jsonFile = "/home/david/inputs/port_inputs.json"
JsonDataLoader loader = new  JsonDataLoader()
loade.parseFile(jsonFile)
```

does this help?

Reply:
οk! now I get what you mean.
you want these operations to happen in the main function.
correct?

Reply:
`public void parseFile(String jsonFile) throws IOException {

        File json = new File(jsonFile);

        JsonFactory factory = new JsonFactory();
        ObjectMapper mapper = new ObjectMapper(factory);
        JsonNode rootNode = mapper.readTree(json);
`+++
is this it?